Okta.Xamarin/Okta.Xamarin.Test/OidcClientShould.cs

```diff
@@ -5,229 +5,267 @@
 
 using System;
 using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 using Xunit;
+using Okta.Xamarin;
+using Newtonsoft.Json;
 
 namespace Okta.Xamarin.Test
 {
-	public class OidcClientShould
-	{
-		[Fact]
-		public void FailWithInvalidConfig()
-		{
-			Assert.Throws<ArgumentNullException>(() => new OidcClient(new OktaConfig()));
-		}
+    public class OidcClientShould
+    {
+        [Fact]
+        public void FailWithInvalidConfig()
+        {
+            Assert.Throws<ArgumentNullException>(() => new OidcClient(new OktaConfig()));
+        }
+
+        [Fact]
+        public void GetCorrectAuthUrl()
+        {
+            OidcClient client = new OidcClient(new OktaConfig("testoktaid", "https://dev-00000.oktapreview.com", "com.test:/redirect+&TEST!@url%20Encode#*^(0)", "com.test:/logout") { Scope = "test hello test_scope" });
```

Comment:
Can we also add a test to verify that the sign-out URL is well-formed ? It should be pretty similar to this test.

Reply:
I'll add that to my list of todos.